src/swsssdk/scripts/sonic-db-cli

```diff
@@ -5,53 +5,55 @@ import swsssdk
 
-def ping_unix_path_single_instance(inst_info):
+def handle_single_instance(op, use_unix_socket, inst_info):
     inst_hostname = inst_info['hostname']
-    inst_unix_socket_path = inst_info['unix_socket_path']
-    r = redis.Redis(host=inst_hostname, unix_socket_path=inst_unix_socket_path)
-    rsp = False
-    msg = 'Could not connect to Redis at {}:{}: Connection refused'.format(inst_hostname, inst_unix_socket_path)
-    try:
-        rsp = r.ping()
-    except redis.exceptions.ConnectionError as e:
-        pass
-    return 'PONG' if rsp else msg
-
-def ping_tcp_single_instance(inst_info):
-    inst_hostname = inst_info['hostname']
-    inst_port = inst_info['port']
-    r = redis.Redis(host=inst_hostname, port=inst_port)
+    if use_unix_socket:
+        inst_unix_socket_path = inst_info['unix_socket_path']
+        r = redis.Redis(host=inst_hostname, unix_socket_path=inst_unix_socket_path)
+    else:
+        inst_port = inst_info['port']
+        r = redis.Redis(host=inst_hostname, port=inst_port)
     rsp = False
-    msg = 'Could not connect to Redis at {}:{}: Connection refused'.format(inst_hostname, inst_port)
+    msg = 'Could not connect to Redis at {}:{}: Connection refused'.format(inst_hostname,
+           inst_unix_socket_path if use_unix_socket else inst_port)
     try:
-        rsp = r.ping()
+        if op == 'PING':
+            rsp = r.ping()
+        elif op == 'SAVE':
+            rsp = r.save()
+        elif op == 'FLUSHALL':
+            rsp = r.flushall()
+        else:
+            msg = "operation {} is not supported".format(op)
     except redis.exceptions.ConnectionError as e:
         pass
-    return 'PONG' if rsp else msg
+    return 'OK' if rsp else msg
 
-def ping_all_instances(namespace, use_unix_socket=False):
-    ping_single_instance = ping_unix_path_single_instance
+def handle_all_instances(namespace, op, use_unix_socket=False):
     # Use the tcp connectivity if namespace is local and unixsocket cmd_option is present.
-    if namespace is None:
-        if not use_unix_socket:
-            ping_single_instance = ping_tcp_single_instance
+    if namespace is not None:
+        use_unix_socket = True
 
     db_insts = swsssdk.SonicDBConfig.get_instancelist(namespace)
-    # ping all redis instances together
-    # TODO: if one of the ping failed, it could fail quickly and not necessary to wait all other pings
+    # operate all redis instances together
+    # TODO: if one of the operation failed, it could fail quickly and not necessary to wait all other pings
     p = Pool(len(db_insts))
-    rsps =  p.map(ping_single_instance, [v for k, v in db_insts.items()])
+    func = partial(handle_single_instance, op, use_unix_socket)
+    rsps = p.map(func, [v for k, v in db_insts.items()])
     msg = []
     for rsp in rsps:
-        if rsp != 'PONG':
+        if rsp != 'OK':
             msg.append(rsp)
     if msg:
         print('\n'.join(msg))
         sys.exit(1)
-    else:
+    elif op == 'PING':
```

Comment:
>elif [](start = 4, length = 4)

Use `if` since it has no relationship with above `if`.

Reply:
DONE